src/eventmaster/api_handlers.go

```diff
@@ -414,37 +414,43 @@ func (h *httpHandler) handleGetDc(w http.ResponseWriter, r *http.Request, _ http
 		h.sendError(w, http.StatusInternalServerError, err, "Error marshalling response to JSON", r.URL.Path)
 		return
 	}
-	h.sendResp(w, "", string(str), r.URL.Path)
+	h.sendResp(w, str, r.URL.Path)
 }
 
-func (h *httpHandler) handleGitHubEvent(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	var info map[string]interface{}
-
-	defer r.Body.Close()
-	reqBody, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error reading request body", r.URL.Path)
-		return
-	}
-	if err := json.Unmarshal(reqBody, &info); err != nil {
-		h.sendError(w, http.StatusBadRequest, err, "Error JSON decoding body of request", r.URL.Path)
-		return
-	}
-
-	id, err := h.store.AddEvent(&UnaddedEvent{
-		Dc:        "github",
-		Host:      "github",
-		TopicName: "github",
-		Data:      info,
-	})
-	if err != nil {
-		h.sendError(w, http.StatusInternalServerError, err, "Error adding event to store", r.URL.Path)
-		return
+func (h *httpHandler) handlePluginEvent(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	pluginName := ps.ByName("name")
+	if plugin, ok := Plugins[pluginName]; ok {
+		defer r.Body.Close()
+		rawData, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			h.sendError(w, http.StatusInternalServerError, err, "Error reading request body", r.URL.Path)
+			return
+		}
+		valid := plugin.ValidateAuth(r, rawData)
+		if !valid {
+			h.sendError(w, http.StatusUnauthorized, errors.New("Invalid signature"), "Error validating request", r.URL.Path)
+			return
+		}
+		events, err := plugin.ParseRequest(r, rawData)
+		if err != nil {
+			h.sendError(w, http.StatusBadRequest, err, "Error parsing event", r.URL.Path)
+			return
+		}
+		var ids []string
+		for _, event := range events {
+			id, err := h.store.AddEvent(event)
+			if err != nil {
+				h.sendError(w, http.StatusInternalServerError, err, "Error adding event to store", r.URL.Path)
+			}
+			ids = append(ids, id)
+		}
+		h.sendObj(w, map[string]interface{}{"event_id": ids}, r.URL.Path)
+	} else {
+		h.sendError(w, http.StatusBadRequest, errors.New("Plugin "+pluginName+" does not exist"), "Error adding event", r.URL.Path)
 	}
-	h.sendResp(w, "event_id", id, r.URL.Path)
 }
 
 func (h *httpHandler) handleHealthCheck(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	// TODO: make this more useful
-	h.sendResp(w, "", "", r.URL.Path)
+	h.sendResp(w, []byte{}, r.URL.Path)
```

Comment:
This should actually check something- at least be wired into shutdown as to fail the HC before stopping :)